pennylane/templates/subroutines/iqp.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the IQP template.
+"""
+from collections import defaultdict
+
+import numpy as np
+
+from pennylane.decomposition import (
+    add_decomps,
+    register_resources,
+    resource_rep,
+)
+from pennylane.operation import Operation
+from pennylane.ops import Hadamard, MultiRZ, PauliRot
+
+
+class IQP(Operation):
+    r"""
+    A template that builds an Instantaneous Quantum Polynomial (IQP) circuit. The gates of these circuits correspond
+    to multi-qubit X rotations, whose generators are given by tensor products of Pauli X operators.
+
+    In this `IQPOpt <https://arxiv.org/pdf/2501.04776>`__ paper and this
+    `train on classical, deploy on quantum <https://arxiv.org/pdf/2503.02934>`__ paper, the authors
+    present methods for analytically approximating expectation values coming from measurements made on IQP circuits.
+    This allows for the classical training of the parameters of these circuits prior to deploying them to a
+    quantum computer for actual computation.
+
+    Certain computational problems such as generative machine learning and combinatorial optimization can be cast as
+    a minimization over functions of these expectation values. Since these circuits are also believed to be hard to
+    sample from using classical algorithms, they can potentially lead to a quantum advantage.
+
```

Comment:
Probably also a reference to the IQP tutorial under a _seealso_ section would be good?